op-e2e/e2eutils/disputegame/helper.go

```diff
@@ -141,6 +143,46 @@ func (h *FactoryHelper) StartAlphabetGame(ctx context.Context, claimedAlphabet s
 	}
 }
 
+func (h *FactoryHelper) StartOutputCannonGame(ctx context.Context, rollupEndpoint string, rootClaim common.Hash) *OutputCannonGameHelper {
+	l2BlockNumber := h.waitForProposals(ctx)
+	l1Head := h.checkpointL1Block(ctx)
+
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	extraData := make([]byte, 64)
+	binary.BigEndian.PutUint64(extraData[24:], l2BlockNumber)
+	binary.BigEndian.PutUint64(extraData[56:], l1Head.Uint64())
```

Comment:
Since we are duplicating this now as in `createCannonGame` and I don't see the `l2BlockNumber` and `l1Head` being used elsewhere, maybe we can merge this in to be something like:

```golang
func (h *FactoryHelper) prepareCannonGame(ctx context.Context) []byte {
	l2BlockNumber := h.waitForProposals(ctx)
	l1Head := h.checkpointL1Block(ctx)
	extraData := make([]byte, 64)
	binary.BigEndian.PutUint64(extraData[24:], l2BlockNumber)
	binary.BigEndian.PutUint64(extraData[56:], l1Head.Uint64())
	return extraData
}
```

Reply:
`l1Head` and `l2BlockNumber` are used in `StartCannonGameWithCorrectRoot`, but we can return all three values and it does clean up some duplication nicely.  Might be coupling the different game types too tightly but it works for now so we can always split back out if need be in the future.